gk/main.c

```diff
@@ -919,11 +1100,32 @@ process_pkts_front(uint8_t port_front, uint8_t port_back,
 				ret = rte_hash_add_key_with_hash(
 					instance->ip_flow_hash_table,
  					&packet.flow, pkt->hash.rss);
-				if (ret < 0) {
+				if (ret == -EINVAL) {
 					RTE_LOG(ERR, HASH,
-						"The GK block failed to add new key to hash table!\n");
+						"The GK block failed to add new key to hash table in %s with invalid parameters!\n",
+						__func__);
 					drop_packet(pkt);
 					continue;
+				} else if (ret == -ENOSPC) {
+					RTE_LOG(WARNING, HASH,
+						"The GK block failed to add new key to hash table in %s due to lack of space!\n",
+						__func__);
+					ret = expand_flow_table(lcore, instance, gk_conf);
+					if (ret < 0) {
+						drop_packet(pkt);
+						continue;
+					}
+
+					ret = rte_hash_add_key_with_hash(
+						instance->ip_flow_hash_table,
+						&packet.flow, pkt->hash.rss);
+					if (ret < 0) {
+						RTE_LOG(ERR, HASH,
+							"The GK block failed to add new key to the expanded hash table in %s!\n",
+							__func__);
+						drop_packet(pkt);
+						continue;
+					}
```

Comment:
I think this might be one of those rare instances where we want to include a `goto` that jumps back up the function, since we're doing the exact same operation once the table is expanded.